mobile_verifier/src/heartbeats.rs

```diff
@@ -52,150 +32,184 @@ pub struct HeartbeatReward {
     pub reward_weight: Decimal,
 }
 
-#[derive(Default)]
-pub struct Heartbeats {
-    pub heartbeats: HashMap<HeartbeatKey, HeartbeatValue>,
+impl From<HeartbeatKey> for HeartbeatReward {
+    fn from(value: HeartbeatKey) -> Self {
+        Self {
+            hotspot_key: value.hotspot_key,
+            cbsd_id: value.cbsd_id,
+            reward_weight: value.cell_type.reward_weight(),
+        }
+    }
 }
 
-/// Minimum number of heartbeats required to give a reward to the hotspot.
-pub const MINIMUM_HEARTBEAT_COUNT: usize = 12;
+pub struct HeartbeatDaemon {
+    pool: sqlx::Pool<sqlx::Postgres>,
+    config_client: Client,
+    heartbeats: Receiver<FileInfoStream<CellHeartbeatIngestReport>>,
+    file_sink: FileSinkClient,
+}
 
-impl Heartbeats {
-    pub async fn validated(exec: impl sqlx::PgExecutor<'_>) -> Result<Self, sqlx::Error> {
-        #[derive(sqlx::FromRow)]
-        pub struct HeartbeatRow {
-            hotspot_key: PublicKeyBinary,
-            cbsd_id: String,
-            reward_weight: Decimal,
-            hours_seen: [bool; 24],
+impl HeartbeatDaemon {
+    pub fn new(
+        pool: sqlx::Pool<sqlx::Postgres>,
+        config_client: Client,
+        heartbeats: Receiver<FileInfoStream<CellHeartbeatIngestReport>>,
+        file_sink: FileSinkClient,
+    ) -> Self {
+        Self {
+            pool,
+            config_client,
+            heartbeats,
+            file_sink,
         }
-
-        let heartbeats = sqlx::query_as::<_, HeartbeatRow>("SELECT * FROM heartbeats")
-            .fetch_all(exec)
-            .await?
-            .into_iter()
-            .map(|hb| {
-                (
-                    HeartbeatKey {
-                        hotspot_key: hb.hotspot_key,
-                        cbsd_id: hb.cbsd_id,
-                    },
-                    HeartbeatValue {
-                        reward_weight: hb.reward_weight,
-                        hours_seen: hb.hours_seen,
-                    },
-                )
-            })
-            .collect();
-        Ok(Self { heartbeats })
     }
 
-    pub fn into_rewardables(self) -> impl Iterator<Item = HeartbeatReward> + Send {
-        self.heartbeats
-            .into_iter()
-            .filter(|(_, value)| value.heartbeat_count() >= MINIMUM_HEARTBEAT_COUNT)
-            .map(|(key, value)| HeartbeatReward {
-                hotspot_key: key.hotspot_key,
-                cbsd_id: key.cbsd_id,
-                reward_weight: value.reward_weight,
-            })
+    pub async fn run(mut self, shutdown: triggered::Listener) -> anyhow::Result<()> {
+        tokio::spawn(async move {
+            let cache = Arc::new(Cache::<(String, DateTime<Utc>), ()>::new());
+
+            let cache_clone = cache.clone();
+            tokio::spawn(async move {
+                cache_clone
+                    .monitor(4, 0.25, std::time::Duration::from_secs(60 * 60 * 3))
```

Comment:
this sets the eviction check to 3 hours but the items in the cache are set to a ttl of 2 hours.  I would suggest bringing the eviction check down to something like 30 mins